osmaxx-py/osmaxx/excerptexport/services/conversion_api_client.py

```diff
@@ -83,37 +85,44 @@ def create_job(self, extraction_order, callback_host):
                 logging.error('Could not retrieve api job id from response.', response)
         return response
 
-    def download_result_files(self, extraction_order):
+    def _download_result_files(self, extraction_order, job_status):
         """
         Downloads the result files if the conversion was finished,
         stores the files into the private storage and attaches them as output files to the extraction order
 
         Args:
             extraction_order: an ExtractionOrder object to attach the output files
-
-        Returns:
-            True if the job status was fetched successful
-            False if it failed
+            job_status: the job status from the conversion api
         """
-        self.login()
-        job_status = self.job_status(extraction_order)
-        if job_status and job_status['status'] == 'done' and job_status['progress'] == 'successful':
-            for download_file in job_status['gis_formats']:
-                if download_file['progress'] == 'successful':
-                    result_response = self.authorized_get(download_file['result_url'])
-                    output_file = OutputFile.objects.create(
-                        mime_type='application/zip',
-                        file_extension='zip',
-                        content_type=download_file['format'],
-                        extraction_order=extraction_order
-                    )
-
-                    file_name = str(output_file.public_identifier) + '.zip'
-                    output_file.file = private_storage.save(file_name, ContentFile(result_response.content))
-                    output_file.save()
-            return True
-        else:
-            return False
+        with transaction.atomic():
+            extraction_order.refresh_from_db()
+            if extraction_order.download_status == extraction_order.DOWNLOAD_STATUS_UNKNOWN:
+                extraction_order.download_status = extraction_order.DOWNLOAD_STATUS_DOWNLOADING
+                extraction_order.save()
+                success = True
+                for download_file in job_status['gis_formats']:
+                    if download_file['progress'] == 'successful':
+                        self._download_file(download_file, extraction_order)
```

Comment:
Our current naming makes this condition confusing. It downloads successfully generated (and thus available on the conversion service) files. However, it reads somewhat as if already successfully downloaded files are being downloaded.